src/pipx/venv.py

```diff
@@ -212,14 +248,19 @@ def upgrade_packaging_libraries(self, pip_args: List[str]) -> None:
         else:
             # TODO: setuptools and wheel? Original code didn't bother
             # but shared libs code does.
-            self.upgrade_package_no_metadata("pip", pip_args)
+            if self.backend != "uv":
+                self.upgrade_package_no_metadata("pip", pip_args)
 
     def uninstall_package(self, package: str, was_injected: bool = False):
         try:
             logger.info("Uninstalling %s", package)
             with animate(f"uninstalling {package}", self.do_animation):
-                cmd = ["uninstall", "-y"] + [package]
-                self._run_pip(cmd)
+                if self.installer != "uv":
```

Comment:
I'd say, explicit is better than implicit.

```suggestion
                if self.installer == "pip":
```